### PR TITLE
Fix content issues in security guides

### PR DESCRIPTION
**Pull request by rolfedh**
This PR incorporates QE feedback from the review of the product security guides into the community files. The changes include:

- Adding additional conditionalization to specific sections.
- Making minor content tweaks and fixing typos.
- Preventing duplicate ID errors that cause build failures.

### CODE REVIEW

File: docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:643]: Hi @rolfedh What is the idea behind `-a`, `-b` suffixes ? It can become tricky to follow/sync across the docs

Comment (rolfedh) [reply docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:643]: Hi @sberyozkin. Thanks for asking. When the community guides are integrated into a product guide, their IDs occupy the same namespace, leading to duplicate ID errors that cause build failures. The simplest way to address this issue is to directly modify the ID values, as shown here. 

Alternatively, a more universal approach involves defining a unique `:context:` attribute (such as the root filename) at the top of the guide and appending `_{context}` to all ID definitions. This method also requires appending `_{context}` to cross-references (xrefs) that use those IDs within the same guide, and appending the literal value of `_{context}` to xrefs or links that use those IDs outside the guide. While I had used a variant of this more complex approach in the previous iteration of the security guides, I am now aiming to simplify it. I'm open to any other suggestions.

Comment (sberyozkin) [reply docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:643]: Thanks @rolfedh for the explanation,  but I have to admit that now having to add these suffixes upstream to various references looks like a workaround which is a bir hard to make systematic.  If that older approach already worked,  why push this extra convention to the upstream docs ?

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:1352]: Why `b` and not a ?

File: docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc
Comment (michalvavrik) [reply docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:643]: So grok is that IDs become duplicate because you mix guides into one while we don't. Problem is that:

- not everyone (I dare say "most people") will not remember or know about this convention
- noone in community can know which IDs may become duplicate because they don't have internal knowledge of product docs

Maybe this https://github.com/asciidoctor/asciidoctor/issues/1696#issuecomment-183873652 could be acceptable (adapted to your use case). I cannot decide this, IMHO you need to include here more people interested in docs with merging power like Max, Guillaume etc.

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (rolfedh) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:1352]: This cross reference in `security-oidc-bearer-token-authentication.adoc` points to the section in `security-oidc-code-flow-authentication.adoc` whose ID is `oidc-request-filters-b`.

Search results for "oidc-request-filters":

```
/asciidoc/security-oidc-bearer-token-authentication.adoc:
  1348:  [[oidc-request-filters-a]]
  1352:  information, see xref:security-oidc-code-flow-authentication#oidc-request-filters-b[OIDC request filters].


/asciidoc/security-oidc-code-flow-authentication.adoc:
  311:  [[oidc-request-filters-b]]


/asciidoc/security-openid-connect-client-reference.adoc:
  1073:  [[oidc-request-filters-c]]
```

File: docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:643]: >noone in community can know which IDs may become duplicate because they don't have internal knowledge of product docs

I can imagine so.

>Maybe this https://github.com/asciidoctor/asciidoctor/issues/1696#issuecomment-183873652 could be acceptable (adapted to your use case)

How would it look upstream, for example, what would we have to type upstream to replace  `[[user-info]]` ?

Comment (michalvavrik) [reply docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:643]: I would say docs experts no this better than I do, but:

- what this PR does ATM is basically https://docs.asciidoctor.org/asciidoc/latest/attributes/id/#as-an-inline-anchor
- while what we need is https://docs.asciidoctor.org/asciidoc/latest/sections/id-prefix-and-separator/ so that it can be customizable downstream

I'd like Rolfe to find the solution.

Comment (michalvavrik) [reply docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:643]: For example, I am quite not sure how the referencing works, like, do you use substitution in ref? people will mess this up. Let's leave it to Rolfe...

File: docs/src/main/asciidoc/security-oidc-code-flow-authentication-tutorial.adoc
Comment (rolfedh) [reply docs/src/main/asciidoc/security-oidc-code-flow-authentication-tutorial.adoc:269]: This cross reference in `security-oidc-code-flow-authentication-tutorial.adoc` points to the section in `security-oidc-code-flow-authentication.adoc` whose ID is `integration-testing-keycloak-devservices-b`.

Search results for "oidc-request-filters":

```
/asciidoc/security-oidc-bearer-token-authentication.adoc:
  768:  [[integration-testing-keycloak-devservices-a]]
  870:  Use <<integration-testing-keycloak-devservices-a,Dev Services for Keycloak>> instead of `KeycloakTestResourceLifecycleManager`


/asciidoc/**security-oidc-code-flow-authentication-tutorial.adoc**:
  269:  **security-oidc-code-flow-authentication.adoc#integration-testing-keycloak-devservices-b**[Dev Services for Keycloak] in
  273:  **security-oidc-code-flow-authentication.adoc#integration-testing-keycloak-devservices-b**[Dev Services for Keycloak] section.


/asciidoc/security-oidc-code-flow-authentication.adoc:
  1801:  **[[integration-testing-keycloak-devservices-b]]**
  1843:  are encouraged to do it with <<integration-testing-keycloak-devservices-b,Dev Services for Keycloak>>.
```

Comment (michalvavrik) [reply docs/src/main/asciidoc/security-oidc-code-flow-authentication-tutorial.adoc:269]: I think just the fact we are having this conversation suggest we shouldn't do it. Why is not id something sensible, like `code-flow-integration-testing-keycloak-devservices` and `bearer-integration-testing-keycloak-devservices`?

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:1352]: I'm sorry Rolfe, but I'm confused, perhaps we can try what Michal suggested below ?

File: docs/src/main/asciidoc/security-oidc-code-flow-authentication-tutorial.adoc
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-oidc-code-flow-authentication-tutorial.adoc:269]: >Why is not id something sensible, like code-flow-integration-testing-keycloak-devservices and bearer-integration-testing-keycloak-devservices?

IMHO it will avoid the confusion

File: docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc
Comment (rolfedh) [reply docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:643]: When considering best practices for guide creation, it’s important to address whether duplicating content across multiple guides is advisable. Doing so contradicts the [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). I recommend consolidating the information in a single location and using cross-references to direct users to that content.

However, let’s set that aside for now and focus on the current situation. We have two options:

1. **Differentiate Conflicting IDs Only When Necessary**: As we've done in this PR, we can address conflicts by differentiating only the IDs that actually conflict. Most users won’t need to be aware of this convention. When new duplicate IDs appear, the documentation team can simply file a PR to resolve them. This method allows us to ignore the majority of IDs that never come into conflict because they don’t appear in the same downstream product guide.

2. **Apply a General Solution**: This method, similar to what is commonly used in Red Hat product documentation, involves setting a `:context:` attribute at the top of the file with the root filename as its value. Then, you append `_{context}` to every ID in the file. Every cross-reference to that ID would then look like `xref:<filename>#<id-value>_<context-value>`. I’m against this solution because it is over-engineered for the scale of the problem and places an unnecessary burden on the community repository. It’s less intuitive and more challenging for new contributors to grasp: even experienced writers frequently make errors with this approach.

Additionally, keep in mind that Red Hat is rolling out a new documentation platform over the next few quarters. I hope this platform will make this issue obsolete and allow us to automate solutions to this problem entirely downstream.

File: docs/src/main/asciidoc/security-oidc-code-flow-authentication-tutorial.adoc
Comment (rolfedh) [reply docs/src/main/asciidoc/security-oidc-code-flow-authentication-tutorial.adoc:269]: YES. If that works for you, let's do this instead!

File: docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc
Comment (michalvavrik) [reply docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc]: Thanks for info!

> I’m against this solution because it is over-engineered for the scale of the problem and places an unnecessary burden on the community repository. It’s less intuitive and more challenging for new contributors to grasp: even experienced writers frequently make errors with this approach.

I don't share this opinion, it feels logical to me, but other suggestions are probably better and I'd like to avoid introducing something just because of Red Hat if little later it becomes unnecessary.

> Additionally, keep in mind that Red Hat is rolling out a new documentation platform over the next few quarters. I hope this platform will make this issue obsolete and allow us to automate solutions to this problem entirely downstream.

okay

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (rolfedh) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc]: That’s perfectly understandable, @sberyozkin. These IDs can trip up many people. Let’s work on making it clearer. I’ve applied @michalvavrik's suggestion to the current PR. Please take a look and let me know if I can make any further tweaks to improve clarity.

File: docs/src/main/asciidoc/security-oidc-code-flow-authentication-tutorial.adoc
Comment (rolfedh) [reply docs/src/main/asciidoc/security-oidc-code-flow-authentication-tutorial.adoc]: I’ve applied @michalvavrik's suggestion to the current PR. Please take a look and let me know if I can make any further tweaks to improve clarity.

File: docs/src/main/asciidoc/security-keycloak-authorization.adoc
Comment (michalvavrik) [inline docs/src/main/asciidoc/security-keycloak-authorization.adoc:229]: we are on 25 now

Comment (michalvavrik) [reply docs/src/main/asciidoc/security-keycloak-authorization.adoc:229]: I wonder if Keycloak authorization API is backwards compatible, because if yes, it's alright.

File: docs/src/main/asciidoc/security-oidc-code-flow-authentication-tutorial.adoc
Comment (michalvavrik) [reply docs/src/main/asciidoc/security-oidc-code-flow-authentication-tutorial.adoc]: ids lgtm, thx

**Comment by sberyozkin** [comment]
@rolfedh LGTM too, thanks, should be ready to go once the build issue is fixed; also makes sense to refer to the 25.0.0 as the minimum KC version as noticed by Michal.
Thanks

File: docs/src/main/asciidoc/security-keycloak-authorization.adoc
Comment (rolfedh) [reply docs/src/main/asciidoc/security-keycloak-authorization.adoc]: https://github.com/quarkusio/quarkus/pull/41791

Comment (rolfedh) [reply docs/src/main/asciidoc/security-keycloak-authorization.adoc]: @michalvavrik  I'm handling it separately in PR https://github.com/quarkusio/quarkus/pull/41791.  Please let me know which version is valid for 3.8. Thank you.

Comment (sberyozkin) [reply docs/src/main/asciidoc/security-keycloak-authorization.adoc]: @rolfedh 24.0.0 or later

**Review by sberyozkin — APPROVED** [review]
LGTM, thanks @rolfedh and @michalvavrik. I'd like to CC @maxandersen and @gsmet just in case before merging it, for them be aware of the proposed convention to  use document name specific prefixes/qualifiers for the document local sections, in order to avoid conflicts in a large single RHBQ doc

**Comment by rolfedh** [comment]
> LGTM, thanks @rolfedh and @michalvavrik. I'd like to CC @maxandersen and @gsmet just in case before merging it, for them be aware of the proposed convention to use document name specific prefixes/qualifiers for the document local sections, in order to avoid conflicts in a large single RHBQ doc

Hi @sberyozkin. Thanks for suggesting this. 
Explanation: When multiple community guides are integrated into a single product guide, their IDs occupy the same namespace. If those IDs are the same, the docs build toolchain [fails the build](https://jenkins.dxp.redhat.com/job/CCS/job/ccs-mr-preview/80536/artifact/pantheon/security-oidc-authentication/ccutil-output.txt). The simplest way to address this issue is to modify only the IDs that produced errors  so they are unique, as was done in this PR.

File: docs/src/main/asciidoc/security-keycloak-authorization.adoc
Comment (rolfedh) [reply docs/src/main/asciidoc/security-keycloak-authorization.adoc]: Handled by:

- https://github.com/quarkusio/quarkus/pull/41791
- https://github.com/quarkusio/quarkus/pull/41792

**Review by michalvavrik — APPROVED** [review]

File: docs/src/main/asciidoc/security-getting-started-tutorial.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-getting-started-tutorial.adoc:46]: Please don't do that.

In the website, I use the following trick:
```
:quickstarts-clone-url: -b 3.8 https://github.com/quarkusio/quarkus-quickstarts.git
```
and that will be an issue with this change.

I think you could do the same. I define it in `_attributes-local.adoc`.

**Review by gsmet — DISMISSED** [review]

File: docs/src/main/asciidoc/security-getting-started-tutorial.adoc
Comment (gsmet) [reply docs/src/main/asciidoc/security-getting-started-tutorial.adoc:46]: To be thorough, here is the entire section:

```
:quickstarts-clone-url: -b 3.8 https://github.com/quarkusio/quarkus-quickstarts.git
:quickstarts-archive-url: https://github.com/quarkusio/quarkus-quickstarts/archive/3.8.zip
:quickstarts-blob-url: https://github.com/quarkusio/quarkus-quickstarts/blob/3.8
:quickstarts-tree-url: https://github.com/quarkusio/quarkus-quickstarts/tree/3.8
```

Comment (rolfedh) [reply docs/src/main/asciidoc/security-getting-started-tutorial.adoc:46]: Thanks for the clarification @gsmet. I should have looked up that attribute value.
cc: @michalvavrik

Comment (rolfedh) [reply docs/src/main/asciidoc/security-getting-started-tutorial.adoc:46]: Hi, @gsmet 

The current value in `asciidoc/_attributes.adoc` is:
```
:quickstarts-clone-url: {quickstarts-base-url}.git
```
Should we update it to the following?
```
:quickstarts-clone-url: -b {github-branch} {quickstarts-base-url}.git
```

Comment (rolfedh) [reply docs/src/main/asciidoc/security-getting-started-tutorial.adoc]: I've removed the change, above. Thanks.

Comment (rolfedh) [reply docs/src/main/asciidoc/security-getting-started-tutorial.adoc]: https://github.com/quarkusio/quarkus/issues/41817

Comment (michalvavrik) [reply docs/src/main/asciidoc/security-getting-started-tutorial.adoc]: not sure if it's just me, but https://github.com/quarkusio/quarkus/issues/41817 is very technical. the gist of this change was: if you read 3.8 branch, guides suggest you to checkout and test main branch of quickstarts. right now, users are following 3.8 guides but using updated code on the main branch

Comment (gsmet) [reply docs/src/main/asciidoc/security-getting-started-tutorial.adoc]: @michalvavrik are you talking about community? If so please be specific. 3.8 guides should point to the 3.8 branch as shown in https://quarkus.io/version/3.8/guides/validation .

Comment (michalvavrik) [reply docs/src/main/asciidoc/security-getting-started-tutorial.adoc]: > @michalvavrik are you talking about community? If so please be specific. 

I was explained this line written by Rolfe:

> I created the following issue to reflect @michalvavrik's feedback 

And that he created #41817 because of me. I never requested this. I wanted that 3.8 branch of documentation referts to 3.8 of quickstarts. that wasn't true for a product and Rolfe wrote that this comes from the community and tried to change it here. But I don't see it in the 3.8 community.

This comment thread is about `git clone {quickstarts-clone-url} -b {github-branch}` on main, which is IMHO unnecessary if community 3.8 is correct.

In that light, I tried to say that I don't understand neither #41817 nor discussion above. As this comment thread and PR are closed, let's forget about it.

Thank you

Comment (rolfedh) [reply docs/src/main/asciidoc/security-getting-started-tutorial.adoc]: One of the QE reviewers requested a change in the community `/asciidoc` files that had been synchronized to the product repository. I was following through to ensure this request didn't get overlooked. Upon inspecting the community attributes files, I didn't see an affordance for the version number. When I reviewed the community `3.12` docs, I noticed that the links pointed to `main` instead of `3.12`. However, had I inspected the `3.8` docs, I would have seen that the links correctly pointed to `3.8`. Thank you for the conversation.

**Comment by rolfedh** [comment]
Prequalified the synchronization job by using the "Try ingesting from an upstream pull request
" pipeline schedule to create https://gitlab.cee.redhat.com/quarkus-documentation/quarkus/-/jobs/23659885

**Pull request merged**
